lib/rules/prefer-object-has-own.js

```diff
@@ -87,6 +111,13 @@ module.exports = {
                                 return null;
                             }
 
+                            const tokenJustBeforeNode = sourceCode.getTokenBefore(node, { includeComments: true });
+
+                            // for https://github.com/eslint/eslint/pull/15346#issuecomment-991417335
+                            if (tokenJustBeforeNode && isProblematicToken(tokenJustBeforeNode) && !sourceCode.isSpaceBetween(node, tokenJustBeforeNode)) {
```

Comment:
We're using [`astUtils.canTokensBeAdjacent()`](https://github.com/eslint/eslint/blob/eafaf52464c01072fcf7be2bd2d91cd0b3ae67d0/lib/rules/utils/ast-utils.js#L1744) for this purpose.

An example:

https://github.com/eslint/eslint/blob/eafaf52464c01072fcf7be2bd2d91cd0b3ae67d0/lib/rules/no-implicit-coercion.js#L241-L250

Reply:
Updated to code to use `astUtils.canTokensBeAdjacent() ` 👍🏻